sklearn/utils/validation.py

```diff
@@ -527,17 +533,16 @@ def column_or_1d(y, warn=False):
 
     """
     shape = np.shape(y)
-    if len(shape) == 1:
-        return np.ravel(y)
     if len(shape) == 2 and shape[1] == 1:
         if warn:
             warnings.warn("A column-vector y was passed when a 1d array was"
                           " expected. Please change the shape of y to "
                           "(n_samples, ), for example using ravel().",
                           DataConversionWarning, stacklevel=2)
-        return np.ravel(y)
-
-    raise ValueError("bad input shape {0}".format(shape))
+        y = np.ravel(y)
+    elif len(shape) != 1:
+        raise ValueError("bad input shape {0}".format(shape))
```

Comment:
You should also mention what you expected as valid input shape (expected 1 dim array-like datastructure).